src/sagas/walletConnect.js

```diff
@@ -0,0 +1,528 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ *  ┌──────────────────────┐     ┌────────┐     ┌───────────┐
+ *  │                      │     │        │     │   SETUP   │
+ *  │ START_WALLET_SUCCESS ├─────►  INIT  ├─────►           │
+ *  │                      │     │        │     │ LISTENERS │
+ *  └──────────────────────┘     └────────┘     └───────────┘
+ *
+ *  ┌──────────────┐    ┌─────────────────┐
+ *  │              │    │                 │
+ *  │ DAPP REQUEST ├────► SESSION_REQUEST │
+ *  │              │    │                 │
+ *  └──────────────┘    └─────────────────┘
+ *
+ *  ┌──────────────┐     ┌────────────────┐
+ *  │              │     │                │
+ *  │ RESET_WALLET ├─────► CLEAR_SESSIONS │
+ *  │              │     │                │
+ *  └──────────────┘     └────────────────┘
+ *
+ *  ┌──────────────┐   ┌──────────────────┐
+ *  │              │   │                  │
+ *  │ URI_INPUTTED ├─┬─► SESSION_PROPOSAL │
+ *  │              │ │ │                  │
+ *  └──────────────┘ │ └──────────────────┘
+ *                   │
+ *  ┌────────────┐   │
+ *  │            │   │
+ *  │ QR_SCANNED ├───┘
+ *  │            │
+ *  └────────────┘
+ *
+ * SESSION_REQUEST: Handles new messages published on the cloud message queue
+ * for the current session by the dApp.
+ * SESSION_PROPOSAL: Handles a new dApp connection, initialized by the pair method
+ * on web3wallet
+ * RESET_WALLET: This action is dispatched when the user resets his wallet.
+ * START_WALLET_SUCCESS: This action is dispatched when the wallet is successfully
+ * loaded.
+ */
+
+import {
+  call,
+  fork,
+  take,
+  all,
+  put,
+  cancel,
+  cancelled,
+  takeLatest,
+  takeLeading,
+  takeEvery,
+  select,
+  race,
+} from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { get, values } from 'lodash';
+
+import { WalletConnectModalTypes } from '../components/WalletConnect/WalletConnectModal';
+import {
+  WALLET_CONNECT_PROJECT_ID,
+  WALLET_CONNECT_FEATURE_TOGGLE,
+  NETWORK,
+} from '../constants';
+import {
+  types,
+  setWalletConnect,
+  setWalletConnectModal,
+  setWalletConnectSessions,
+  onExceptionCaptured,
+  setWCConnectionFailed,
+} from '../actions';
+import { checkForFeatureFlag, showPinScreenForResult } from './helpers';
+
+const AVAILABLE_METHODS = {
+  HATHOR_SIGN_MESSAGE: 'hathor_signMessage',
+};
+const AVAILABLE_EVENTS = [];
+
+/**
+ * Those are the only ones we are currently using, extracted from
+ * https://docs.walletconnect.com/2.0/specs/clients/sign/error-codes
+ */
+const ERROR_CODES = {
+  UNAUTHORIZED_METHODS: 3001,
+  USER_DISCONNECTED: 6000,
+  USER_REJECTED: 5000,
+  USER_REJECTED_METHOD: 5002,
+  INVALID_PAYLOAD: 5003,
+};
+
+const Core = class {};
```

Comment:
Maybe it's better to write a comment here

Reply:
Added the comment in 5dda8b0f7e539095fac758e1497def8e7beb3340